client/actions/requestHandlers/handleCreateRecipe.js

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import upload from './handleFileUpload';
+import { ADD_NEW_RECIPE } from '../types';
+
+
+function addNewRecipe(recipe) {
+  return {
+    type: ADD_NEW_RECIPE,
+    recipe
+  };
+}
+
+export default data => (dispatch) => {
+  if (typeof (data.image) === 'object') {
+    upload(data.image).end((err, res) => {
+      if (!err || res.ok) {
+        data.image = res.body.url;
+        Materialize.toast('Recipe Successfully Created!', 4000);
+        axios.post('/api/recipes', data).then((res) => {
+          dispatch(addNewRecipe(res.data.Details));
+        }).catch(err => Materialize.toast('Recipe was not created!', 4000));
+      } else {
+        Materialize.toast('Unable to load image', 4000);
+      }
+    });
+  } else {
+    axios.post('/api/recipes', data).then((res) => {
```

Comment:
I tried this method but i was getting an error. I think the reason is because is because node dispatches both the image upload and API call asynchronously.